daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/TypedValueGenerators.scala

```diff
@@ -84,7 +84,10 @@ object TypedValueGenerators {
 
       override def t: Type = TypeNumeric(scale)
 
-      override def inj[Cid]: Numeric => Value[Cid] = ValueNumeric
+      override def inj[Cid]: Numeric => Value[Cid] = { n =>
+        require(n.scale() == scale)
```

Comment:
I am not sure how to handle this, but properly typed `Numeric` should have the good scale.